From baba68cdcb44fc11d0ba8ce2c13eb5b06bbd9b33 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Wed, 30 Oct 2019 07:15:23 +0100
Subject: [PATCH] keymap.h: add missing includes to fix musl build

Needed to fix these build errors:

In file included from keymap.c:13:0:
keymap.h:23:2: error: unknown type name 'u_int32_t'
  u_int32_t scancode;

keymap.h:36:1: error: unknown type name 'error_t'
 error_t parse_keymap(char *fname, struct keymap **keymap, bool verbose);

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
Patch was not sent upstream due to broken mailing list setup:

<majordomo@vger.kernel.org>: host vger.kernel.org[209.132.180.67] said: 553
    5.7.1 Hello [xx.xx.xx.xx], for your MAIL FROM address
    <bernd.kuhls@t-online.de> policy analysis reported: Your address is not
    liked source for email (in reply to MAIL FROM command)

Hello [xx.xx.xx.xx], for your MAIL FROM address <berndkuhls@hotmail.com>
 policy analysis reported: Your address is not liked source for email 

 utils/common/keymap.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/utils/common/keymap.h b/utils/common/keymap.h
index f2b29632..bb1edce9 100644
--- a/utils/common/keymap.h
+++ b/utils/common/keymap.h
@@ -1,4 +1,8 @@
 /* SPDX-License-Identifier: GPL-2.0 */
+
+#include <argp.h>
+#include <sys/types.h>
+
 #ifndef __KEYMAP_H
 #define __KEYMAP_H
 
diff -uNr v4l-utils-1.18.0.orig/utils/ir-ctl/keymap.h v4l-utils-1.18.0/utils/ir-ctl/keymap.h
--- v4l-utils-1.18.0.orig/utils/ir-ctl/keymap.h	2019-09-22 11:22:54.000000000 +0200
+++ v4l-utils-1.18.0/utils/ir-ctl/keymap.h	2019-10-30 07:06:18.250548011 +0100
@@ -1,4 +1,8 @@
 /* SPDX-License-Identifier: GPL-2.0 */
+
+#include <argp.h>
+#include <sys/types.h>
+
 #ifndef __KEYMAP_H
 #define __KEYMAP_H
 
diff -uNr v4l-utils-1.18.0.orig/utils/keytable/keymap.h v4l-utils-1.18.0/utils/keytable/keymap.h
--- v4l-utils-1.18.0.orig/utils/keytable/keymap.h	2019-09-22 11:22:54.000000000 +0200
+++ v4l-utils-1.18.0/utils/keytable/keymap.h	2019-10-30 07:06:56.218816126 +0100
@@ -1,4 +1,8 @@
 /* SPDX-License-Identifier: GPL-2.0 */
+
+#include <argp.h>
+#include <sys/types.h>
+
 #ifndef __KEYMAP_H
 #define __KEYMAP_H
 
-- 
2.20.1
